File: dot_product/simple_loop.jl
using Random
using LinearAlgebra


function dot_them(u, v, num_runs)
    for i=1:num_runs
        dot(u, v)
    end
end


function many_dot_products(vec_size=10^4, num_runs=10^6)
    u = rand(vec_size)
    v = rand(vec_size)
    @timed dot_them(u, v, num_runs)
end


function main()
    run_time = many_dot_products()[2]
    print(run_time)
end


main()

File: dot_product/simple_loop.py
import time
import random


VEC_SIZE = 1000
NUM_RUNS = 10**6


u = [random.random() for i in range(VEC_SIZE)]
v = [random.random() for i in range(VEC_SIZE)]


start = time.time()
for i in range(NUM_RUNS):
    sum(map(lambda x, y: x * y, u, v))
end = time.time()
print(f"{end - start}")
